London Class 9 - Karleen Richards - HTML/CSS Coursework - Week 1

karleenmsrichards opened the pull request:
<!--

The title for your pull request should be made in this format

CITY CLASS_NO - FIRST_NAME LAST_NAME - MODULE - WEEK_NO

For example,

London Class 7 - Chris Owen - HTML/CSS - Week 1

Please complete the details below this message

-->

**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name:
- Your City:
- Your Slack Name:

# Homework Details

- Module:
- Week:

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

karleenmsrichards committed:
Initial requirements added to web page

karleenmsrichards committed:
Added summary tags correctly

karleenmsrichards committed:
Added images and fixed structure of html

karleenmsrichards committed:
added logo and header of the webpage

karleenmsrichards committed:
changes made to main video and nav bar aligned centered

karleenmsrichards committed:
changed position of the articles and added color to the background of <summary>

karleenmsrichards committed:
Update style.css

karleenmsrichards committed:
Fixed articles columns

karleenmsrichards committed:
Final changes made

Added all the requirements for HTML/CSS Coursework Week 1 - 3 articles with article:firstchild

karleenmsrichards committed:
Structured html/css

Also changed the title

karleenmsrichards committed:
Rename elements and containers

Fixed HTML sematic elements and CSS names

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+
+main {
+    margin-bottom: 50px;
+
```
Being constant about line gaps will make the code easier to read. There is no need for this one here.

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+
+.border-bottom {
+    border-bottom: 1px solid black;
+}
```
Try to keep indention consistent throughout your code

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+      margin-bottom: 50px;
+}
+
+ul li a {
```
This will target all a tags within uls on your site using a class to just target the menu could be better

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+}
+
+.h1 {
+     font-size: 30px;
```
Setting font sizes as pixels will break one of the requirements of accessibility using rem here would be better

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+     justify-content: center;
+     padding-right: 68px;
+     text-decoration: underline;
+}
```
It looks like you have done some really good experiments of flexbox throughout this file! After you get the effect you want it is worth disabling the attributes in the browsers inspector to check what they are actually doing. In this instance `display: flex`, `flex-direction: row;` and `justify-content: center;` will have no effect because they are properties to set on a parent element to affect its children and this element doesnt have any children

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+ }
+
+ .top-right-p1 {
+     padding: 5px 0 0 100px;
```
The layout you selected is surprisingly difficult to create, although easier now with css grid, positioning elements with large padding values may work well at a single screen size but will likely break on different screen sizes so is bet to be avoided

PeteLindsell replied:
It is alway good to change the width of the browser while developing to see what happens

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+    padding-left: 100px;
+}
+
+.aside-border {
+    border: 1px solid rgb(180, 175, 175);
+    margin-bottom: 40px;
+    padding: 15px;
+    font-weight: bolder;  
+}
```
You could add this code to .aside-container preventing the need for an extra element to add a border

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+    display: flex;
+    flex-direction: column;
+    flex: 0;
+    order: 2;
```
Very impressive to use the order property to answer the brief requirement of targeting the first child while also getting the look you wanted! In practice order is best avoided as it confuses the document flow for somebody tabbing through the page but impressive here

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+    margin-right: 10px;
+}
+
+.article2 {
+    display: flex;
+    flex-direction: column;
+    flex: 2;
+    order: 1;
+    padding: 25px;
+}
+
+.article3 {
+    display: flex;
+    flex: 2;
+    flex-direction: column;
+    order: 3;
+    padding: 25px; 
+}
```
If you have lots of repeated code you can set that all with a generic selector such as `article` or create a class `home-page-articles` eg: 
```
article {
    display: flex;
    flex: 2;
    flex-direction: column;
    padding: 25px;
}
```
and then just set the unique properties and values to the individual elements

PeteLindsell commented on css/style.css:
```diff
@@ -6,3 +6,228 @@
+/* Footer styles*/
+
+footer {
+    left: 0;
+    bottom: 0;
```
These are not doing anything here. The footer is looking offset to the right because you are setting `width: 100%;` which the browser takes and then adds the padding to it. if you remove width the footer will naturaly fill the space

PeteLindsell commented on index.html:
```diff
@@ -6,17 +6,115 @@
+        <section class="top-right">
+        <p class="top-right-p1">BRITAIN</p>
+        <p class="top-right-p2">SUBSCRIBE</p>
+        <img class="hburger-img" alt="Picture of a hamburger menu" src="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQhhwlrve66B8GHVhBRth6kDr6ml2SQR73R5g&usqp=CAU">
```
Alt text is difficult you generally dont want to describe the image instead state its function. Here something like "Menu" or "Display main menu" would be more suitable

PeteLindsell commented on index.html:
```diff
@@ -6,17 +6,115 @@
+        <aside class="aside-container">
+          
+            <h3 class="stories">MAIN HEADLINES</h3>
+            <details>
+              <summary class="aside-border">Top outerwear this winter</summary>
+            </details>
```
the detail and summary tag are used as follows
```
<details>
    <summary>Details</summary>
    Something small enough to escape casual notice.
</details>
```
The summary is always displayed and the rest of the content of details is hidden until the summary is clicked

PeteLindsell commented on index.html:
```diff
@@ -6,17 +6,115 @@
+
+      </div>
+
+      <h3 class="stories">FASHION NEWS</h3>
+
+      <section class="article-container">
+        <article>
+          <h1>Styling tips for the puffer</h1>
```
You should only have 1 h1 on a page which uniquely titles the content of the page. After a h1 you can have one or more h2 but you should not skip to h3 or h4

PeteLindsell commented on index.html:
```diff
@@ -6,17 +6,115 @@
+          <h1>Styling tips for the puffer</h1>
+          <p>There are various ways to style the puffer jacket but there are different rules which depends on numerous
+            factors. What's the length, what's the fit and what about the color. <strong>Author: Mishy Mish</strong></p>
+          <button><a class="link"
```
A link inside a button is not valid. A link should always have a href linking to part of a page or another url. A button will fire some other event such as submitting a form or opening a modal. Links are often designed to look like button but this shoul be done with css rather than a button element

PeteLindsell reviewed (commented):
You have created a really good looking design which is surprisingly difficult to accomplish. I have made quite a few comments which I am happy to discuss any that dont make sense. 

This is really good for your first week of html css!

karleenmsrichards committed:
Added a <a href> to an element

karleenmsrichards committed:
Fixed semantic tags in html

Made changes to <section> tags to <div> tags.

karleenmsrichards committed:
Corrections made to HTML/CSS

1. Adjusted h tags by followings numerical increments
2. Fixed alt attributes for images to state function etc.
3. Removed button with link inside and added styles in CSS for links.
Changed some semantic tags.
4. Used fixed font sizes, used fixed colour attributes.
